packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -209,9 +212,18 @@ class FuchsiaDevices extends PollingDeviceDiscovery {
       return null;
     }
     final String name = words[1];
-    final String resolvedHost = await _fuchsiaSdk.fuchsiaDevFinder.resolve(
-      name,
-    );
+    String resolvedHost;
+    // Uses ffx disovery workflow by default. The legacy device-finder
+    // workflow can be enabled by setting the environment variable FUCHSIA_DISABLED_ffx_discovery=1.
+    final String disabledFfxDiscovery = _platform.environment.containsKey('FUCHSIA_DISABLED_ffx_discovery')
+      ? _platform.environment['FUCHSIA_DISABLED_ffx_discovery'] : '0';
+    if (disabledFfxDiscovery == '1') {
+      resolvedHost = await _fuchsiaSdk.fuchsiaDevFinder.resolve(
```

Comment:
Why do we have `listDevices` on the _fuchsiaSdk object but we have `resolve` on the specific tools?

Reply:
It seems like tech debt, I added a TODO and will fix in a follow up CL